cmake/CubCudaConfig.cmake

```diff
@@ -1,132 +1,24 @@
-  # CUB_CUDA_FLAGS_RDC: Additional CUDA flags for targets compiled with RDC.
-  # CUB_CUDA_FLAGS_NO_RDC: Additional CUDA flags for targets compiled without RDC.
-  #
-  # This is necessary because CUDA SMs 5.3, 6.2, and 7.2 do not support RDC, but
-  # we want to always build some targets (e.g. testing/cuda/*) with RDC.
-  # We work around this by building the "always RDC" targets without support for
-  # those SMs. This requires two sets of CUDA_FLAGS.
-  #
-  # Enabling any of those SMs along with the ENABLE_RDC options will result in a
-  # configuration error.
-  #
-  # Because of how CMake handles the CMAKE_CUDA_FLAGS variables, every target
-  # generated in a given directory will use the same value for CMAKE_CUDA_FLAGS,
-  # which is determined at the end of the directory's scope. This means caution
-  # should be used when trying to build different targets with different flags,
-  # since they might not behave as expected. This will improve with CMake 3.18,
-  # which add the DEVICE_LINK genex, fixing the issue with using per-target
-  # CUDA_FLAGS: https://gitlab.kitware.com/cmake/cmake/-/issues/18265
-  set(CUB_CUDA_FLAGS_BASE "${CMAKE_CUDA_FLAGS}")
-  set(CUB_CUDA_FLAGS_RDC)
-  set(CUB_CUDA_FLAGS_NO_RDC)
-
-  # Archs that don't support RDC:
-  set(no_rdc_archs 53 62 72)
-
-  # Find the highest arch:
-  list(SORT all_archs)
-  list(LENGTH all_archs max_idx)
-  math(EXPR max_idx "${max_idx} - 1")
-  list(GET all_archs ${max_idx} highest_arch)
-
-  set(option_init OFF)
-  if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
-    set(option_init ON)
-  endif()
-  option(CUB_DISABLE_ARCH_BY_DEFAULT
-    "If ON, then all compute architectures are disabled on the initial CMake run."
-    ${option_init}
-  )
-
-  set(option_init ON)
-  if (CUB_DISABLE_ARCH_BY_DEFAULT)
-    set(option_init OFF)
-  endif()
-
-  set(arch_flags)
-  set(num_archs_enabled 0)
-  foreach (arch IN LISTS all_archs)
-    option(CUB_ENABLE_COMPUTE_${arch}
-      "Enable code generation for sm_${arch}."
-      ${option_init}
-    )
-
-    if (CUB_ENABLE_COMPUTE_${arch})
-      math(EXPR num_archs_enabled "${num_archs_enabled} + 1")
-
-      if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
-        if (NOT ${num_archs_enabled} EQUAL 1)
-          message(FATAL_ERROR
-            "NVC++ does not support compilation for multiple device architectures "
-            "at once."
-          )
-        endif()
-        set(arch_flag "-gpu=cc${arch}")
-      elseif ("Clang" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
-        set(arch_flag "--cuda-gpu-arch=sm_${arch}")
-      else()
-        set(arch_flag "-gencode arch=compute_${arch},code=sm_${arch}")
-      endif()
-
-      string(APPEND arch_message " sm_${arch}")
-      string(APPEND CUB_CUDA_FLAGS_NO_RDC " ${arch_flag}")
-      if (NOT arch IN_LIST no_rdc_archs)
-        string(APPEND CUB_CUDA_FLAGS_RDC " ${arch_flag}")
-      endif()
-    endif()
-  endforeach()
+if (NOT CUB_IN_THRUST)
+  message(FATAL_ERROR
```

Comment:
Since we install this file it means we will break `find_package(cub)` for any consumers

Reply:
No consumers should be including this file other than the Thrust build. It doesn't really need to be installed, and is only used to configure the tests, etc. on dev/ci builds.